src/app/header/menu/menu.component.ts

```diff
@@ -1,33 +1,57 @@
 import { Component, OnInit, Input } from '@angular/core';
-import { Router, ActivatedRoute } from '@angular/router';
+import { ActivatedRoute } from '@angular/router';
+
 import { ContentfulService } from '../../core/contentful.service';
+import { Menu, MenuItem } from '../../models/menu.model';
 
 @Component({
   selector: 'asm-menu',
   templateUrl: './menu.component.html',
   styleUrls: ['./menu.component.scss']
 })
 export class MenuComponent implements OnInit {
-  @Input() menu: any[];
+  @Input() menu: [Menu | MenuItem];
   event: any;
+  openedSection: string;
 
-  constructor(private router: Router,
+  constructor(
     private contentful: ContentfulService,
     private route: ActivatedRoute) { }
 
   ngOnInit() {
     this.event = this.route.snapshot.data.event;
   }
 
-  getUrl(item) {
+  getUrl(item: any) {
     return this.contentful.getUrl(item);
   }
 
-  isMenuSectioned(item) {
+  isMenuSectioned(item: Menu) {
     return item.__typename === 'Menu' && item.items.some(subitem => subitem.__typename === 'Menu');
   }
 
-  onNavigation(item, event: Event) {
+  onNavigation(item: any, event: Event) {
     this.contentful.onNavigation(item, event);
   }
+
+  openSection(item: Menu) {
+    this.openedSection = item.label;
+  }
+
+  closeSection(item: Menu) {
```

Comment:
This function is not used in template. Remove or change close logic to use this.